src/factories.spec.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-env jest */
+import { simple, keyValue } from './factories'
+
+describe('factories', () => {
+  it('should use options of object type', () => expect(simple.object('object').options).toMatchSnapshot())
+  it('should use options of specific string type', () => expect(simple.string('string').options).toMatchSnapshot())
+  it('should use options of specific array type', () => expect(simple.array('array').options).toMatchSnapshot())
+  it('should use options of specific boolean type', () => expect(simple.bool('boolean').options).toMatchSnapshot())
+  it('should use options of specific keyValue type', () => expect(keyValue('keyValue').options).toMatchSnapshot())
+
+  it('should use overwrited options of object type', () => expect(simple.object({ name: 'object', defaultData: { o: 'test' } }).options).toMatchSnapshot())
```

Comment:
Your PR  is not about options but about defaultData (and in a higher level, about isInitialized)
1. Make isInitialized bugguy again
2. Find a test that show it bugs
3. Fix this test

Only that